### PR TITLE
Use modules instead of plugins

### LINKED ISSUE #21: Use modules instead of plugins with Play 2.4
Use the new Play 2.4 dependency injection with modules instead of plugins.

https://playframework.com/documentation/2.4.0-M2/ScalaDependencyInjection#Play-libraries

### PR DESCRIPTION
**Pull request by Mogztter**
* Use modules
 * Update sample project to use version 3.0.0-M1-SNAPSHOT

Resolves #21

### CODE REVIEW

File: src/main/scala/play/api/libs/mailer/MailerPlugin.scala [hunk 1]
```diff
@@ -142,33 +65,54 @@ trait MailerAPI extends MailerJavaAPI {
...
 
-abstract class CommonsMailer(smtpHost: String, smtpPort: Int,
-                            smtpSsl: Boolean,
-                            smtpTls: Boolean,
-                            smtpUser: Option[String],
-                            smtpPass: Option[String],
-                            debugMode: Boolean,
-                            smtpTimeout: Option[Int],
-                            smtpConnectionTimeout: Option[Int]) extends MailerAPI {
+abstract class STMPMailer(smtpHost: String, smtpPort: Int,
```
Comment (jroper) [inline src/main/scala/play/api/libs/mailer/MailerPlugin.scala]: STMPMailer -> SMTPMailer

Comment (Mogztter) [inline src/main/scala/play/api/libs/mailer/MailerPlugin.scala]: Ooops nice catch :smile:

**Comment by jroper** [comment]
Travis build failed.

**Comment by Mogztter** [comment]
Ooops, Intellij rename feature add a `$` sign but I don't know why...and  I missed one :cry:

**Comment by Mogztter** [comment]
Build passed \o/

**Pull request merged**